addons/auth_signup/models/res_users.py

```diff
@@ -224,6 +239,71 @@ def send_unregistered_user_reminder(self, after_days=5):
             template = self.env.ref('auth_signup.mail_template_data_unregistered_users').with_context(dbname=self._cr.dbname, invited_users=invited_users[user])
             template.send_mail(user, email_layout_xmlid='mail.mail_notification_light', force_send=False)
 
+    def _should_alert_new_device(self):
+        if self._mfa_type() is None:
+            return True
+        if self._mfa_type() != 'totp_mail':
+            cookies = request.httprequest.cookies
+            key = cookies.get('td_id')
+            if key:
+                is_verified_device = request.env['auth_totp.device']._check_credentials_for_uid(
+                    scope="browser", key=key, uid=self.id)
+                if not is_verified_device:
+                    return True
+        return False
+
+    def _alert_new_device(self):
+        for user in self.env.user:
+            if user.email:
+                template = self.env.ref('auth_signup.alert_new_device_email')
```

Comment:
What happens if the template has been removed ? Should we skip, raise, ... ? Currently it will break. Don't know what functionally you want to do.